crates/protocol/src/proto_envelope.rs

```diff
@@ -11,7 +11,7 @@ use warg_crypto::{hash::DynHashError, signing, Decode, Signable};
 /// The envelope struct is used to keep around the original
 /// bytes that the content was serialized into in case
 /// the serialization is not canonical.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
```

Comment:
Likewise, the change should not be necessary (and this type is intentionally not serializable).